homeassistant/components/eps/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for EPS integration."""
+import logging
+
+from pyepsalarm import EPS
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_PASSWORD, CONF_TOKEN, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_TOKEN): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+async def _get_device_site(hass: core.HomeAssistant, token, username, password):
+    eps_api = EPS(token, username, password)
+    return await hass.async_add_executor_job(eps_api.get_site)
+
+
+class EPSConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for EPS."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        site = None
+
+        if user_input is None:
```

Comment:
This function could be simplified by changing this to if user_input is not None, always passing errors into show form (if it is an empty dict ntohing happens). That way you only call show_Form once.

Reply:
Fixed, thanks for the idea